### PR TITLE
Add network-drive extension

### PR DESCRIPTION
**Pull request by SuoweiHu**
## Description

I created an extension to simplify mounting and dismounting network drives in Finder. 
I hope you will like it (・ω・)b !!!!  

## Video Demo

https://share.cleanshot.com/lFYpNh3cbDmX07QnPVjF

## Screencast

network-drive-1
network-drive-2

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/network-drive/src/utils-preference.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { getPreferenceValues } from "@raycast/api";
+interface Preferences {
+  pref_smb_ip: string;
+  pref_smb_usr: string;
+  pref_smb_pwd: string;
+}
```
Comment (pernielsentikaer) [inline extensions/network-drive/src/utils-preference.tsx:6]:
```suggestion
```
This is already included in raycast-env.d.ts

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Is there a way to add a timeout, seems like if I enters a wrong domain it never stops loading

network-drive 2024-06-11 at 11 54 50


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by SuoweiHu** [comment]
Thank you for the suggested changes @pernielsentikaer,
I'll implement those changes very soon !

**Comment by SuoweiHu** [comment]
Feature's up ! ( ⸝⸝´꒳`⸝⸝)

https://github.com/raycast/extensions/assets/35317549/10953177-83e9-40b2-99b9-3727000377e7

**Comment by pernielsentikaer** [comment]
Hi 👋

Could you check the update I made that I didn't break something 🙂

**Comment by SuoweiHu** [comment]
That's **exactly** what I am looking for !!!
 
(p.s. I am not exactly familiar with the async programming, I spend like almost an hour trying to implement this but it keeps failing to work as I anticipated !  

Tons of thanks to you @pernielsentikaer

**Comment by pernielsentikaer** [comment]
You should be able to pull my contributions with `npx @raycast/api@latest pull-contributions`

**Review by pernielsentikaer — APPROVED** [review]
Let's get the first version out to see how it goes 🙂

**Pull request merged**